storage/src/adb/any/sync/client.rs

```diff
@@ -179,11 +175,19 @@ where
 {
     /// Create a new sync client
     pub(crate) async fn new(config: Config<E, K, V, H, T, R>) -> Result<Self, Error> {
+        Self::new_with_updater(config, None).await
+    }
+
+    /// Create a new sync client whose target can be updated
+    pub(crate) async fn new_with_updater(
```

Comment:
What do you think about just supporting "updater"? I'm not sure there will ever be a case where we sync to a static target...

Reply:
I guess a sync to some historical root could qualify for the static root world, so maybe makes sense to keep around as its a pretty minimal thing anyways..

Reply:
I'm just going to move the updater into the sync `Config`